src/ResourceManager/Compute/Commands.Compute/Generated/GalleryImage/GalleryImageCreateOrUpdateMethod.cs

```diff
@@ -0,0 +1,547 @@
+                        }
+                        galleryImage.PurchasePlan.Publisher = this.PurchasePlanPublisher;
+                    }
+
+                    if (this.MyInvocation.BoundParameters.ContainsKey("PurchasePlanProduct"))
+                    {
+                        if (galleryImage.PurchasePlan == null)
+                        {
+                            galleryImage.PurchasePlan = new ImagePurchasePlan();
+                        }
+                        galleryImage.PurchasePlan.Product = this.PurchasePlanProduct;
+                    }
+
+                    var result = GalleryImagesClient.CreateOrUpdate(resourceGroupName, galleryName, galleryImageName, galleryImage);
+                    var psObject = new PSGalleryImage();
+                    ComputeAutomationAutoMapperProfile.Mapper.Map<GalleryImage, PSGalleryImage>(result, psObject);
+                    WriteObject(psObject);
+                }
+            });
+        }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceManager.Common.ArgumentCompleters.ResourceGroupCompleter()]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        public string GalleryName { get; set; }
+
+        [Alias("Name")]
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Position = 2,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        public string ImageDefinitionName { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            Position = 3,
+            ValueFromPipelineByPropertyName = true)]
+        [LocationCompleter("Microsoft.Compute/Galleries")]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            Position = 4,
+            ValueFromPipelineByPropertyName = true)]
+        public string Publisher { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            Position = 5,
+            ValueFromPipelineByPropertyName = true)]
+        public string Offer { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            Position = 6,
+            ValueFromPipelineByPropertyName = true)]
+        public string Sku { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            Position = 7,
+            ValueFromPipelineByPropertyName = true)]
+        public OperatingSystemStateTypes OsState { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            Position = 8,
```

Comment:
@hyonholee 8 is too many positional params, we ideally recommend having under 4. Can you take out the position for the params?